src/include/stir/listmode/ListEventRecordMapFromFile.h

```diff
@@ -0,0 +1,83 @@
+/* ListEventRecordMapFromFile.h
+ Read List-Mode Event Data using map from file: Header File
+ Jannis Fischer
+ jannis.fischer@cern.ch
+
+	Copyright 2015 ETH Zurich, Institute of Particle Physics
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+ */
+
+/*!
+
+  \file
+  \ingroup listmode
+  \brief Declaration of class stir::CListEventRecordMapFromFile
+
+  \author Jannis Fischer
+*/
+
+#include <fstream>
+#include <string>
+#include <vector>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/unordered_map.hpp>
+
+#include "stir/CartesianCoordinate3D.h"
+#include "stir/DetectionPosition.h"
+
+#ifndef __local_SAFIR_ListEventRecordMapFromFile_H__
+#define __local_SAFIR_ListEventRecordMapFromFile_H__
+
+START_NAMESPACE_STIR
+
+//! Class providing map functionality to convert detector indices to spatial coordinates. 
```

Comment:
need \ingroup. say that it's used by SAFIR.
I would put the doc on the file format in the main doxygen here as opposed to the function below.

I think this class needs renaming? Not ListEventRecordMapFromFile but something like DetectorCoordinateMapFromFile?